pilot/pkg/xds/internalgen.go

```diff
@@ -88,6 +107,15 @@ func (sg *InternalGen) OnDisconnect(con *Connection) {
 	// Note that it is quite possible for a 'connect' on a different istiod to happen before a disconnect.
 }
 
+func (sg *InternalGen) Run(stop <-chan struct{}) {
+	if sg.Store != nil {
```

Comment:
Should just `&&` these - the cleanupQueue uses the store and the periodic cleanup just puts things on the queue.